clash-prelude/src/Clash/Promoted/Nat.hs

```diff
@@ -560,3 +611,15 @@ leToPlusKN
   -> r
 leToPlusKN r = r @(n - k)
 {-# INLINE leToPlusKN #-}
+
+-- | Change a function that has an argument with an @(n <= 0)@ constraint to a
+-- function with an argument that has an @(n ~ 0)@ constraint.
+leZeroToEqZero :: forall n r. (KnownNat n, n <= 0) => (n ~ 0 => r) -> r
```

Comment:
As per discussion in chat I think this function should not exist currently due to the bug in the natnormalise plugin: https://github.com/clash-lang/ghc-typelits-natnormalise/issues/85

Essentially if you have `n <= 0` constraint you are on the way to mount doom of bugs currently. Having this function indicates that this will work at all which is the wrong message to send now. So I would prefer to keep this function out until that bug is fixed.

Reply:
I removed the function again. Thanks for the feedback and the discussion.